app/models/solidus_bolt/gateway.rb

```diff
@@ -69,8 +69,26 @@ def void(response_code, gateway_options)
       ActiveMerchant::Billing::Response.new(false, e, {})
     end
 
-    def credit(_amount, _response_code, _gateway_options)
-      raise NotImplementedError, 'credit method has not been implemented in SolidusBolt::Gateway class'
+    def credit(amount, response_code, _gateway_options)
+      payment = Spree::Payment.find_by(response_code: response_code)
+      payment_source = payment.source
+      payment_method = payment.payment_method
+      currency = payment.currency
+
+      SolidusBolt::Transactions::RefundService.call(
+        transaction_reference: response_code,
+        amount: amount,
+        currency: currency,
+        payment_method: payment_method
+      )
+
+      ActiveMerchant::Billing::Response.new(
+        true,
+        'Transaction Refunded', payment_source.attributes,
+        authorization: response_code
```

Comment:
<img width="1202" alt="Screenshot 2022-05-13 at 13 00 49" src="https://user-images.githubusercontent.com/387690/168270098-e11beaee-6d47-450a-a6ab-5608a86ac2f9.png">

The reference of the refund is the same as the payment ([3KCN-TWCV-KHBF](https://merchant-sandbox.bolt.com/transaction/3KCN-TWCV-KHBF)), but in the truth, it's [J324-TPYZ-8R9Y](https://merchant-sandbox.bolt.com/transaction/J324-TPYZ-8R9Y) on Bolt.

Reply:
If I change the reference to the Refund then it also changes the reference for the original payment

Reply:
fixed here
[d325465](https://github.com/nebulab/solidus_bolt/commit/d32546569329ae84c6af45756f27db2a4d6e9f42)